kotest-extensions/src/jvmTest/kotlin/com/sksamuel/kt/extensions/system/SystemEnvironmentExtensionTest.kt

```diff
@@ -63,14 +63,21 @@ class SystemEnvironmentExtensionTest : FreeSpec() {
 
 class SystemEnvironmentTestListenerTest : WordSpec() {
 
-   private val setl = SystemEnvironmentTestListener("mop", "dop", mode = OverrideMode.SetOrOverride)
+   private val setl = SystemEnvironmentTestListener(
+      environment = mapOf(
+         "mop" to "dop",
+         "dop" to null,
+      ),
+      mode = OverrideMode.SetOrOverride
+   )
 
    override fun extensions() = listOf(setl)
 
    init {
       "sys environment extension" should {
          "set environment variable" {
             System.getenv("mop") shouldBe "dop"
+            System.getenv("dop") shouldBe null
```

Comment:
🔴 I think this test needs to check that the variable was removed from the environment, I.E. having it configured before the test runs

Otherwise we might catch a false positive here where our implementation does nothing with the variable and thus it stays null as before

Reply:
@LeoColman Apologies for the long delay! The tests are ready for a re-run and I believe I addressed your test concerns here.